### ISSUE TITLE
Exception with Rider EAP 24

### ISSUE THREAD

**Issue by ianbrandt** [hsz/idea-gitignore#385]
Got this generating from Unity.gitignore without duplicates:

```
Isolated exception
kotlin.KotlinNullPointerException
	at com.jetbrains.resharper.daemon.components.RiderMarkupHost$1.invoke(RiderMarkupHost.kt:54)
	at com.jetbrains.resharper.daemon.components.RiderMarkupHost$1.invoke(RiderMarkupHost.kt:34)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:163)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:142)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:154)
	at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.Signal.fire(Signal.kt:27)
	at com.jetbrains.rider.util.reactive.ViewableMap.put(ViewableMap.kt:33)
	at com.jetbrains.resharper.document.DocumentHost.viewEditableEntity$rider(DocumentHost.kt:96)
	at com.jetbrains.resharper.document.DocumentHost$1$1.run(DocumentHost.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:926)
	at com.jetbrains.resharper.document.DocumentHost$1.invoke(DocumentHost.kt:62)
	at com.jetbrains.resharper.document.DocumentHost$1.invoke(DocumentHost.kt:38)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:163)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:142)
	at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:154)
	at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:122)
	at com.jetbrains.rider.util.reactive.Signal.fire(Signal.kt:27)
	at com.jetbrains.rider.util.reactive.ViewableMap.put(ViewableMap.kt:33)
	at com.jetbrains.rider.framework.impl.RdMap$put$1.invoke(RdMap.kt:98)
	at com.jetbrains.rider.framework.base.RdReactiveBase.localChange(RdReactiveBase.kt:45)
	at com.jetbrains.rider.framework.impl.RdMap.put(RdMap.kt:98)
	at com.jetbrains.resharper.document.DocumentHost.registerModelDocument$rider(DocumentHost.kt:289)
	at com.jetbrains.resharper.editors.TextControlHost.bindEditor$rider(TextControlHost.kt:270)
	at com.jetbrains.resharper.editors.TextControlHost.editorCreated(TextControlHost.kt:215)
	at sun.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:99)
	at com.intellij.util.EventDispatcher.access$200(EventDispatcher.java:35)
	at com.intellij.util.EventDispatcher$2.invoke(EventDispatcher.java:79)
	at com.sun.proxy.$Proxy38.editorCreated(Unknown Source)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:200)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:183)
	at mobi.hsz.idea.gitignore.util.Utils.createPreviewEditor(Utils.java:369)
	at mobi.hsz.idea.gitignore.outer.OuterIgnoreWrapper.<init>(OuterIgnoreWrapper.java:119)
	at mobi.hsz.idea.gitignore.outer.OuterIgnoreLoaderComponent$IgnoreEditorManagerListener.fileOpened(OuterIgnoreLoaderComponent.java:163)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:437)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:398)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:380)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:373)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:44)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:230)
	at com.sun.proxy.$Proxy114.fileOpened(Unknown Source)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$null$7(FileEditorManagerImpl.java:938)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5$1.run(FileEditorManagerImpl.java:1031)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.flushRequest(FocusManagerImpl.java:655)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$9(FocusManagerImpl.java:478)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:2632)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:471)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:466)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.doWhenFocusSettlesDown(IdeFocusManagerImpl.java:76)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5.run(FileEditorManagerImpl.java:1028)
	at com.intellij.openapi.util.BusyObject$Impl$Simple.execute(BusyObject.java:123)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.notifyPublisher(FileEditorManagerImpl.java:1024)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl4$8(FileEditorManagerImpl.java:935)
	at com.intellij.util.ui.UIUtil.invokeAndWaitIfNeeded(UIUtil.java:2653)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4(FileEditorManagerImpl.java:959)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl3(FileEditorManagerImpl.java:787)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl2$4(FileEditorManagerImpl.java:769)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:149)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:109)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl2(FileEditorManagerImpl.java:769)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileWithProviders(FileEditorManagerImpl.java:700)
	at com.intellij.openapi.fileEditor.ex.FileEditorManagerEx.openFile(FileEditorManagerEx.java:148)
	at mobi.hsz.idea.gitignore.util.Utils.openFile(Utils.java:188)
	at mobi.hsz.idea.gitignore.util.Utils.openFile(Utils.java:178)
	at mobi.hsz.idea.gitignore.actions.NewFileAction.actionPerformed(NewFileAction.java:115)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:215)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:232)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:309)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:929)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:136)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:299)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:116)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:86)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:109)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:118)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:116)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:513)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:45)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:533)
	at java.awt.Component.processMouseEvent(Component.java:6541)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6306)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:651)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Rider 2017.1 EAP
Build #RS-171.4456.902, built on June 29, 2017
Rider EAP User
Expiration date: July 29, 2017
JRE: 1.8.0_112-release-736-b17 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.12.5

**Comment by hsz** [comment]
@ianbrandt @appsolutegeek Can you please write down some steps to reproduce? I couldn't get this error with Rider 2017.1.1 nor 2017.1

**Comment by ianbrandt** [comment]
@hsz, I can no longer reproduce this error as well.  I can't say if it was the Rider 2017.1.1 update, or the .ignore 2.1.1 update that did it.

**Comment by appsolutegeek** [comment]
Sorry for the delay. i can reproduce it without an issue. I just reopen it. And lower right there is a red exclamation mark. and this is in the event log

I also updated the plugin to latest version i believe which reports. version 2.1.1

Happens every time, using the latest version of rider 2017.1.1.

I am using a Unity project, i can upload somewhere if you like.

25/08/2017
08:13	KotlinNullPointerException: Isolated exception
			--- JAVA_EXCEPTION ---
			kotlin.KotlinNullPointerException
			at com.jetbrains.rider.daemon.components.RiderMarkupHost$1.invoke(RiderMarkupHost.kt:46)
			at com.jetbrains.rider.daemon.components.RiderMarkupHost$1.invoke(RiderMarkupHost.kt:26)
			at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:163)
			at com.jetbrains.rider.util.reactive.IViewableMap$view$2.invoke(Interfaces.kt:122)
			at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:142)
			at com.jetbrains.rider.util.reactive.IViewableMap$view$1.invoke(Interfaces.kt:122)
			at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:154)
			at com.jetbrains.rider.util.reactive.IViewableMap$adviseAddRemove$1.invoke(Interfaces.kt:122)
			at com.jetbrains.rider.util.reactive.Signal.fire(Signal.kt:27)
			at com.jetbrains.rider.util.reactive.ViewableMap.put(ViewableMap.kt:33)
			at com.jetbrains.rider.document.DocumentHost.viewEditableEntity$rider(DocumentHost.kt:97)
			at com.j... (show balloon)

**Comment by appsolutegeek** [comment]
@hsz I find it strange that you don't have the error. What type of project (although should not make a difference) are you using. What OS ?

I am using a MAC with a unity project.

**Comment by ianbrandt** [comment]
@hsz, Ah, I am getting the error on open of a Unity project as well, just as described by @appsolutegeek.  I do not get the error when opening a plain old .Net Core solution.

I'm currently on Unity 5.6.2f1.

**Comment by appsolutegeek** [comment]
ah! @ianbrandt so that must be the difference :-) Must admit, i only use it for Unity so I was unable to see it being fixed on other project types.

**Comment by samodovdi** [comment]
got the same error on Rider 2017.3 right after opening asp.net core 2.0 project

**Comment by samodovdi** [comment]
**environment:**
windows 10 pro(1709, build 16299.192) Rider 2017.3

**step to reproduce the bug:**
1. bootstrap a new solution:
```
dotnet new console -n TestConsole
dotnet new sln -n TestConsole
dotnet sln TestConsole.sln add TestConsole\TestConsole.csproj
```
2. add an empty .gitignore file near the .sln file
3. open solution in rider. Add solution folder (for ex. Solution items)
4. add .gitignore to the solution folder, open .gitignore file and close rider
5. **important step!** init a git repository:
```
git init
```
6. open rider. you will get an exception, if not, try to reopen rider a few more times and/or close/open the .gitignore file from the solution explorer

Test solution:
[Test.zip](https://github.com/hsz/idea-gitignore/files/1696990/Test.zip)

Screenshot after the described steps:
![image](https://user-images.githubusercontent.com/4929842/35831702-bcc8c0f8-0adb-11e8-8724-60af1bb3fa15.png)

**Comment by ssamooilov** [comment]
+1
I just open Rider and open .gitignore file.
[idea.zip](https://github.com/hsz/idea-gitignore/files/1761315/idea.zip)

**Comment by hsz** [comment]
Good news everyone!

I was able to reproduce this issue some time ago... Rider (and only this IDE) thrown a `KotlinNullPointerException` - and it was not possible to catch it in my plugin in any way.
The Preview Editor was rendered and displayed correctly anyway - so we have ended up just with annoying exception notification in IDE...

However - I have digged in the JetBrains core a bit - they have an issue that may be fixed in `2018.1` releases - who knows. But following change (6b65442) fixed this problem:

```diff
--- a/src/mobi/hsz/idea/gitignore/outer/OuterIgnoreWrapper.java
+++ b/src/mobi/hsz/idea/gitignore/outer/OuterIgnoreWrapper.java
@@ -118,7 +118,7 @@ public void linkSelected(LinkLabel aSource, Object aLinkData) {
 
         for (final VirtualFile outerFile : outerFiles) {
             Document document = FileDocumentManager.getInstance().getDocument(outerFile);
-            Editor outerEditor = document != null ? Utils.createPreviewEditor(document, project, true) : null;
+            Editor outerEditor = document != null ? Utils.createPreviewEditor(document, null, true) : null;
 
             if (outerEditor != null) {
                 final JScrollPane scrollPanel = ScrollPaneFactory.createScrollPane(outerEditor.getComponent());
```

Sorry that you had to wait, but finally it's fixed and will be released with `v2.5.0`.
